From 79f02a53316f90543d60269d7c06727c376f423b Mon Sep 17 00:00:00 2001
From: Antoine Martin <dev@ayakael.net>
Date: Wed, 24 Aug 2022 18:41:08 +0000
Subject: [PATCH 1/1] pass RestoreParallelDisable

A bug was introduced in dotnet7 that causes issues with parallel restore
operations when using mono-flavored runtime. A workaround is setting
/p:RestoreParallelDisable=true.

This patch allows proper flow of that flag through the build

See https://github.com/dotnet/runtime/issues/77364

---
diff --git a/repos/Directory.Build.props b/repos/Directory.Build.props
index 04ab89d11d..c33468f711 100644
--- a/repos/Directory.Build.props
+++ b/repos/Directory.Build.props
@@ -144,6 +144,7 @@
     <StandardSourceBuildArgs>$(StandardSourceBuildArgs) /p:ReferencePackageNupkgCacheDir="$(ReferencePackagesDir)"</StandardSourceBuildArgs>
     <StandardSourceBuildArgs>$(StandardSourceBuildArgs) /p:PreviouslySourceBuiltNupkgCacheDir="$(PrebuiltSourceBuiltPackagesPath)"</StandardSourceBuildArgs>
     <StandardSourceBuildArgs>$(StandardSourceBuildArgs) /p:SourceBuildUseMonoRuntime=$(SourceBuildUseMonoRuntime)</StandardSourceBuildArgs>
+    <StandardSourceBuildArgs>$(StandardSourceBuildArgs) /p:RestoreDisableParallel=$(RestoreDisableParallel)</StandardSourceBuildArgs>
 
     <StandardSourceBuildCommand>$(ProjectDirectory)\build$(ShellExtension)</StandardSourceBuildCommand>
   </PropertyGroup>

diff --git a/ArcadeOverrides/SourceBuildArcadeBuild.targets b/ArcadeOverrides/SourceBuildArcadeBuild.targets
index 94d7777c3b..2468132ad3 100644
--- a/ArcadeOverrides/SourceBuildArcadeBuild.targets
+++ b/ArcadeOverrides/SourceBuildArcadeBuild.targets
@@ -80,6 +80,8 @@
       <InnerBuildArgs>$(InnerBuildArgs) /bl:$(CurrentRepoSourceBuildBinlogFile)</InnerBuildArgs>
       <!-- Flow ContinuousIntegrationBuild to the inner build. -->
       <InnerBuildArgs Condition="'$(ContinuousIntegrationBuild)' == 'true'">$(InnerBuildArgs) /p:ContinuousIntegrationBuild=true</InnerBuildArgs>
+      <!-- Flow RestoreDisableParallel to the inner build -->
+      <InnerBuildArgs Condition="'$(RestoreDisableParallel)' == 'true'">$(InnerBuildArgs) /p:RestoreDisableParallel=true</InnerBuildArgs>
 
       <!-- The inner build needs to reference the overall output dir for nupkg transport etc. -->
       <InnerBuildArgs>$(InnerBuildArgs) /p:SourceBuildOutputDir=$(SourceBuildOutputDir)</InnerBuildArgs>
